plugins/filter/listify.py

```diff
@@ -0,0 +1,266 @@
+                    "nameAlias": "",
+                    "ownerKey": "",
+                    "ownerTag": ""
+                }
+            }
+        }
+    ]
+error:
+  description: The error information as returned from the APIC
+  returned: failure
+  type: dict
+  sample:
+    {
+        "code": "122",
+        "text": "unknown managed object class foo"
+    }
+raw:
+  description: The raw output returned by the APIC REST API (xml or json)
+  returned: parse error
+  type: str
+  sample: '<?xml version="1.0" encoding="UTF-8"?><imdata totalCount="1"><error code="122" text="unknown managed object class foo"/></imdata>'
+sent:
+  description: The actual/minimal configuration pushed to the APIC
+  returned: info
+  type: list
+  sample:
+    {
+        "fvTenant": {
+            "attributes": {
+                "descr": "Production environment"
+            }
+        }
+    }
+previous:
+  description: The original configuration from the APIC before the module has started
+  returned: info
+  type: list
+  sample:
+    [
+        {
+            "fvTenant": {
+                "attributes": {
+                    "descr": "Production",
+                    "dn": "uni/tn-production",
+                    "name": "production",
+                    "nameAlias": "",
+                    "ownerKey": "",
+                    "ownerTag": ""
+                }
+            }
+        }
+    ]
+proposed:
+  description: The assembled configuration from the user-provided parameters
+  returned: info
+  type: dict
+  sample:
+    {
+        "fvTenant": {
+            "attributes": {
+                "descr": "Production environment",
+                "name": "production"
+            }
+        }
+    }
+filter_string:
+  description: The filter string used for the request
+  returned: failure or debug
+  type: str
+  sample: ?rsp-prop-include=config-only
+method:
+  description: The HTTP method used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: POST
+response:
+  description: The HTTP response from the APIC
+  returned: failure or debug
+  type: str
+  sample: OK (30 bytes)
+status:
+  description: The HTTP status from the APIC
+  returned: failure or debug
+  type: int
+  sample: 200
+url:
+  description: The HTTP url used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: https://10.11.12.13/api/mo/uni/tn-production.json
+"""
+
+
+def listify(d, *keys):
+    return list(listify_worker(d, keys, 0, {}, ""))
+
+
+def listify_worker(d, keys, depth, cache, prefix):
+    prefix += keys[depth] + "_"
+
+    if keys[depth] in d:
+        for item in d[keys[depth]]:
+            cache_work = cache.copy()
+            if isinstance(item, dict):
+                for k, v in item.items():
+                    if isinstance(v, list) and all(isinstance(x, (str, int, float, bool, bytes)) for x in v):
+                        cache_key = prefix + k
+                        cache_value = ",".join(v)
```

Comment:
Hi @shrsr, I suggest not to join the list values into a string value because [aci_bd_subnet.py ](https://github.com/CiscoDevNet/ansible-aci/blob/ca6110c99741e1af4198ee894b10096f66c3c555/plugins/modules/aci_bd_subnet.py#L415) requires list input in attribute value of scope.

Reply:
@velotiger Our backend aci ansible module for bd_subnet joins the list of values and the backend ansible code here which validates the values: https://github.com/ansible/ansible/blob/05259aa6ce60965852066b3d4e72a943c0175c07/lib/ansible/module_utils/common/parameters.py#L648 parses the values as strings by joining them to compare with the choices. 

That being said I agree that we can combine the if and elseif part in the latest commit which is almost a duplication of code due to the weird behaviour stated above. However, either way all the formats are accepted-> scope: [private, shared] or scope: private,shared or
scope:
  - private
  - shared